profile2/impl/src/java/org/sakaiproject/profile2/dao/impl/ProfileDaoImpl.java

```diff
@@ -45,1155 +35,649 @@
+        query.setParameter(USER_UUID, userId);
+        return query.getResultStream().findAny().orElse(null);
+    }
+
+    @Override
+    public List<ProfileImageUploaded> getOtherProfileImageRecords(final String userId) {
+        TypedQuery<ProfileImageUploaded> query = getSession().createNamedQuery(QUERY_OTHER_PROFILE_IMAGE_RECORDS, ProfileImageUploaded.class);
+        query.setParameter(USER_UUID, userId);
+        return query.getResultList();
+    }
+
+    @Override
+    public ProfileFriend getConnectionRecord(final String userId, final String friendId) {
+        TypedQuery<ProfileFriend> query = getSession().createNamedQuery(QUERY_GET_FRIEND_RECORD, ProfileFriend.class);
+        query.setParameter(USER_UUID, userId);
+        query.setParameter(FRIEND_UUID, friendId);
+        return query.getResultStream().findAny().orElse(null);
+    }
+
+    @Override
+    public CompanyProfile getCompanyProfile(final String userId, final long companyProfileId) {
+        TypedQuery<CompanyProfile> query = getSession().createNamedQuery(QUERY_GET_COMPANY_PROFILE, CompanyProfile.class);
+        query.setParameter(USER_UUID, userId);
+        query.setParameter(ID, companyProfileId);
+        return query.getResultStream().findAny().orElse(null);
+    }
+
+    @Override
+    public GalleryImage getGalleryImageRecord(final String userId, final long imageId) {
+        TypedQuery<GalleryImage> query = getSession().createNamedQuery(QUERY_GET_GALLERY_RECORD, GalleryImage.class);
+        query.setParameter(USER_UUID, userId);
+        query.setParameter(ID, imageId);
+        return query.getResultStream().findAny().orElse(null);
+    }
+
+    @Override
+    public ProfileImageOfficial getOfficialImageRecordForUser(final String userUuid) {
+        TypedQuery<ProfileImageOfficial> query = getSession().createNamedQuery(QUERY_GET_OFFICIAL_IMAGE_RECORD, ProfileImageOfficial.class);
+        query.setParameter(USER_UUID, userUuid);
+        return query.getResultStream().findAny().orElse(null);
+    }
+
+    @Override
+    public boolean addNewConnection(final ProfileFriend profileFriend) {
+        try {
+            getSession().persist(profileFriend);
+            return true;
+        } catch (Exception e) {
+            log.warn("request friend failed, {}", e.toString());
+        }
+        return false;
+    }
+
+    @Override
+    public boolean updateConnection(final ProfileFriend profileFriend) {
+        try {
+            getSession().merge(profileFriend);
+            return true;
+        } catch (Exception e) {
+            log.warn("confirm friend request failed, {}", e.toString());
+        }
+        return false;
+    }
+
+    @Override
+    public boolean removeConnection(final ProfileFriend profileFriend) {
+        try {
+            Session session = getSession();
+            session.remove(session.merge(profileFriend));
```

Comment:
session.remove is ok?

Reply:
yes, it is the jpa version of delete
https://docs.jboss.org/hibernate/orm/5.4/userguide/html_single/Hibernate_User_Guide.html#pc-remove